pkg/collector/metric/utils.go

```diff
@@ -232,3 +232,20 @@ func getCPUArchitecture() (string, error) {
 
 	return "", fmt.Errorf("no CPU power model found for architecture %s", myCPUModel)
 }
+
+func getCPUPackageMap() (cpuPackageMap map[int32]string) {
```

Comment:
Sorry, I didn't track that one but yes this will read the file at the beginning because the topology is not supposed to change over time. The ReadFile function is efficient enough to quickly read a small file and close it. 

Parsing `/proc/cpuinfo` is also another choice but we need to make sure that there will be no formatting compatibility in the future. Also the larger cores the more you need to run through a big file with irrelevant information. That's why I think using the raw file (`/sys/devices/system/cpu/cpu%d/topology/physical_package_id`) is better.